app/src/_map/tracks/tracks.selectors.js

```diff
@@ -0,0 +1,107 @@
+import { createSelector } from 'reselect';
+import { getTemporalExtent } from '../module/module.selectors.js';
+
+export const getTracksData = state => state.map.tracks.data;
+
+const filterGeojsonByTimerange = (geojson, { start, end }) => {
+  if (!geojson || !geojson.features) return null;
+  const featuresFiltered = geojson.features.reduce((acc, feature) => {
+    const hasTimes = feature.properties.coordinateProperties.times && feature.properties.coordinateProperties.times.length > 0;
+    if (hasTimes) {
+      const filtered = feature.geometry.coordinates.reduce((acc2, coordinate, index) => {
+        const timeCoordinate = feature.properties.coordinateProperties.times[index];
+        const isInTimeline = timeCoordinate > start && timeCoordinate < end;
+        if (isInTimeline) {
+          acc2.coordinates.push(coordinate);
+          acc2.times.push(timeCoordinate);
+        }
+        return acc2;
+      }, { coordinates: [], times: [] });
+      if (!filtered.coordinates.length) return acc;
+
+      const filteredFeature = {
+        ...feature,
+        geometry: {
+          ...feature.geometry,
+          coordinates: filtered.coordinates
+        },
+        properties: {
+          ...feature.properties,
+          coordinateProperties: {
+            times: filtered.times
+          }
+        }
+      };
+      acc.push(filteredFeature);
+    }
+    return acc;
+  }, []);
+  const geojsonFiltered = {
+    ...geojson,
+    features: featuresFiltered
+  };
+  return geojsonFiltered;
+};
+
+export const getTracksStyles = createSelector(
+  [getTemporalExtent, getTracksData],
+  (temporalExtent, tracks) => {
+    const hasTemporalExtent = temporalExtent && temporalExtent.length > 0;
+    const hasTracks = tracks && tracks.length > 0;
+    if (!hasTemporalExtent || !hasTracks) return null;
+
+    const timerange = {
+      start: temporalExtent[0].getTime(),
+      end: temporalExtent[1].getTime()
+    };
+    const styles = tracks.reduce((acc, track) => {
+      if (!track.data) return acc;
+
+      const source = `${track.id}Track`;
+      const style = {
+        sources: {
+          [source]: {
+            type: 'geojson',
+            data: filterGeojsonByTimerange(track.data, timerange)
+          }
+        },
+        layers: [
+          {
+            id: `${track.id}Lines`,
+            source,
+            type: 'line',
+            interactive: true,
```

Comment:
This parameter is now deprecated (and ignored) by Mapbox GL

Reply:
Interaction on layers is now handled here https://github.com/GlobalFishingWatch/map-client/blob/as-module/app/src/_map/glmap/Map.container.js#L43

Reply:
Removed, thanks for the info!